src/Uno.UI/UI/Xaml/Internal/ContentRoot.cs

```diff
@@ -88,10 +80,20 @@ public ContentRoot(ContentRootType type, Color backgroundColor, UIElement? rootE
 			case ContentRootType.XamlIslandRoot:
 				FocusAdapter = new FocusManagerXamlIslandAdapter(this);
 				FocusManager.SetFocusObserver(new FocusObserver(this));
+				XamlIslandRoot = (XamlIsland)rootElement!;
 				break;
 			default:
 				throw new InvalidOperationException("Unknown content root type.");
 		}
+
+		//TODO Uno: Does not match WinUI exactly, additional logic can be ported later.
+		VisualTree = new VisualTree(coreServices, backgroundColor, rootElement, this);
+		InputManager = new InputManager(this);
+		_contentRootEventListener = new ContentRootEventListener(this);
+
+		CompositionTarget = new CompositionTarget(this);
+		CompositionTarget.Root = ElementCompositionPreview.GetElementVisual(VisualTree.RootElement);
+		CompositionTarget.Root.CompositionTarget = CompositionTarget;
```

Comment:
Can you avoid having to move this code block below the switch? The order is matching WinUI and I am afraid that it could lead to some unexpected lifecycle issues when we align application startup further with WinUI